Assets/Mirror/Runtime/NetworkManager.cs

```diff
@@ -636,7 +627,7 @@ void Initialize()
 
         void RegisterServerMessages()
         {
-            server.Connected.AddListener(OnServerConnectInternal);
+            server.Authenticated.AddListener(OnServerAuthenticated);
```

Comment:
beautiful!  

This seriously reduces coupling between NM and NS,   it is becoming a lot easier to use NS without NM.